### PR TITLE
Initial Commit, 4 actions and Data ingestion added

### PR DESCRIPTION
**Pull request by cs-adarsh-gupta**
#### UTCs: 
 
- [x] Connector installation verified. 
- [x] Connector logo verified. 
- [x] Actions and Playbook list verified. 
- [x] Playbook tags verified. 
- [x] All Playbook are in info mode verified. 
- [x] All Playbook are in inactive mode verified.

### CODE REVIEW

File: claroty-xdome/info.json [hunk 1]
```diff
@@ -0,0 +1,777 @@
...
+          "required": false,
+          "editable": true,
+          "type": "select",
+          "options": [
+            "Unresolved",
+            "Resolved",
+            "Acknowledged"
+          ],
+          "tooltip": "Specify the Alert Status you want to filter Alerts.",
+          "description": "Specify the device type you want to filter Devices. One or more Device type can be provided in a comma separated format."
```
Comment (cs-harsh-sharma) [inline claroty-xdome/info.json]: please update description

File: claroty-xdome/info.json [hunk 2]
```diff
@@ -0,0 +1,777 @@
...
+          "type": "multiselect",
+          "options": [
+            "Level 1",
+            "Level 2",
+            "Level 3",
+            "Level 4",
+            "Level 5"
+          ],
+          "tooltip": "Specify the device purdue level you want to filter Devices. One or more Purdue levels can be provided in a comma separated format.",
+          "description": "Specify the device purdue level you want to filter Devices. One or more Purdue levels can be provided in a comma separated format."
```
Comment (cs-harsh-sharma) [inline claroty-xdome/info.json]: please update the description and tooltip

File: claroty-xdome/info.json [hunk 3]
```diff
@@ -0,0 +1,777 @@
...
+          "editable": true,
+          "type": "multiselect",
+          "options": [
+            "Very Low",
+            "Low",
+            "Medium",
+            "High",
+            "Critical"
+          ],
+          "tooltip": "Specify the Known Vulnerabilities you want to filter devices. One or more Known Vulnerabilities can be provided in a comma separated format.",
```
Comment (cs-harsh-sharma) [inline claroty-xdome/info.json]: please update the description and tooltip

File: claroty-xdome/info.json [hunk 4]
```diff
@@ -0,0 +1,777 @@
...
+          "editable": true,
+          "type": "multiselect",
+          "options": [
+            "Low",
+            "Medium",
+            "High",
+            "Critical"
+          ],
+          "tooltip": "Specify the Risk Score you want to filter devices. One or more risk score can be provided in a comma separated format.",
+          "description": "Specify the Risk Score you want to filter devices. One or more risk score can be provided in a comma separated format."
```
Comment (cs-harsh-sharma) [inline claroty-xdome/info.json]: please update the description and tooltip

File: claroty-xdome/info.json [hunk 5]
```diff
@@ -0,0 +1,777 @@
...
+        },
+        {
+          "title": "Limit",
+          "name": "limit",
+          "visible": true,
+          "required": false,
+          "editable": true,
+          "type": "integer",
+          "value": 100,
+          "tooltip": "Specify the maximum amount of Devices to fetch. Default value is 100",
```
Comment (cs-harsh-sharma) [inline claroty-xdome/info.json]: please update the description and tooltip

File: claroty-xdome/info.json [hunk 6]
```diff
@@ -0,0 +1,777 @@
...
+          "required": false,
+          "editable": true,
+          "type": "json",
+          "value": [
+            {
+              "field": "published_date",
+              "order": "desc"
+            }
+          ],
+          "tooltip": "Specify how the returned data should be sorted, If more than one sort clause is passed, additional clauses will be used to sort data that is equal in all previous clauses. Default value is [{\"field\":\"uid\",\"order\":\"asc\"}]",
```
Comment (cs-harsh-sharma) [inline claroty-xdome/info.json]: please update the description and tooltip

File: claroty-xdome/operations.py [hunk 7]
```diff
@@ -0,0 +1,234 @@
...
+class ClarotyXDOMEConnector:
+    def __init__(self, config):
+        self.server_url = config.get('server_url')
+        if not (self.server_url.startswith('https://') or self.server_url.startswith('http://')):
+            self.server_url = 'https://' + self.server_url
+        self.server_url = self.server_url.strip('/')
+        self.api_key = config.get('api_key')
+        self.verify_ssl = config.get('verify_ssl')
+
+    def make_request(self, endpoint, method='GET', data=None, params=None, files=None):
```
Comment (cs-harsh-sharma) [inline claroty-xdome/operations.py]: Almost all methods are of type "POST" so please update the default value of method from "GET" to "POST"

File: claroty-xdome/operations.py [hunk 8]
```diff
@@ -0,0 +1,234 @@
...
+                from connectors.debug_utils.curl_script import make_curl
+                make_curl(method, endpoint, headers=headers, params=params, data=data, verify_ssl=self.verify_ssl)
+            except Exception as err:
+                logger.error(f"Error in curl utils: {str(err)}")
+
+            response = requests.request(method, url, params=params, files=files, data=data, headers=headers,
+                                        verify=self.verify_ssl)
+            if response.ok:
+                logger.info('Successfully got response for url {}'.format(url))
+                if method.upper() == 'DELETE':
```
Comment (cs-harsh-sharma) [inline claroty-xdome/operations.py:44]: Does this line executes in any case/scenario?

File: claroty-xdome/playbooks/playbooks.json [hunk 9]
```diff
@@ -0,0 +1,1147 @@
...
...IRupMmTVq4cOHt27e/9eAhhFJ65MgRKCkqKgLz9u3bYcLSRolpSooAbNEURb8gEZfMRE3TRRzQzKGHmV0wVUkxM+w6pboZVKjLxNuJSOxyTFBVoc5S9D8FPBUVFYjDNWvWYAoyGdHR1tZmAR89ehSenzdvHhySnZ09depUZLW1+oSEBJ/PN7zWiMBX6Dqdzv8CLElSenp6WVnZ5cuXQY6Qxp0RYF0g4qsJrgR05J76Qffi+N4EhTZp3uLavr313ndu2AaTxuUlpr2TO25tht8uxRUmm8rwO1l2znenpGft2bMHYRY1Zq3eshqt0tZNoJ46dQolx2azobty5UpENSKTRAQ5jJuHDx9GzLtcLngGq7Tmrlu37sCBA+3t7VbNQ12sra1dvXp1VHmU84X9Rarj5rRp00afFxIRRYdT8BQ45DNByRPTd+WrhF9P7n+/XvnUP3QdB5P+PHXQXi00vvFH6Ew9lONck0kokxgV3a7y3x5/++3ivLw8KJVlGVG3fv36xYsXwwB2dOPGjQ4c5BE5f/48SkhHRweAMQzriI2NxYDy8nKUU0QvxqCYFRQUoHHo0KEdO3YsWLAA3kYXGXj9+nWsPisra2BgwOPx7N+/H6UrCmyhRhsjxw+lFvYL/hf3lO2VmcHSYkN/aLGFgv4a3bbQnViaFc7Re3396SWZPYukmCsDNn+KPENLOJpr2gSbIOI1lBGWlpb61g+LsFDEmBVIs2bNQo5BL06XnJwcnFjIVcQnNuXRo0fYCxwYljcgOH5QS3FF0CqKgvHTp0/HU8DggMEATLTUIofRRvrMnTsXtIgOK7wtToTMtIh8M7mwlUh4WIk+IoaJlauqbnSf9IT3NkthF2UhIdfuLst+Vt3tqvH1L4t3f/DMm61+5/evu78Xh48IIshj+WsJ+ciUkKQmlWQ+FWTl5w8Ceiy9YwRag9gJs9YZ6nycdjjH/YM0ZYIdMaIKom1Mv1oaBB4T3ITaYvRXSye8UpnreEMUlQGB4RXSR+3+5O9P0N7NsE90GkTSmGJTxbH9ecgMXaP40A0Tpgg6GRJVmRFZ1TWdBdSwTJ0ulxaQHAqqrsFwiKkCsVNxLAPz/0tzYA7MgTkwB+bAHJgDc2AOzIE5MAfmwByYA3NgDsyBOTAH5sAcmANzYA7MgTnwyyD/EmAAwlClWE9meAMAAAAASUVORK5CYII=",
```
Comment (cs-harsh-sharma) [inline claroty-xdome/playbooks/playbooks.json]: update value of image key to null

File: claroty-xdome/operations.py (hunk 8 above)
Comment (cs-adarsh-gupta) [reply claroty-xdome/operations.py:44]: Yes, In case of Delete operations, Not currently used in connector, But in future versions It can be helpful

**Review by cs-harsh-sharma — APPROVED** [review]

**Pull request merged**